frame/system/src/lib.rs

```diff
@@ -450,6 +448,19 @@ pub mod pallet {
 			storage::unhashed::kill_prefix(&prefix);
 			Ok(().into())
 		}
+
+		/// Make some on-chain remark and emit event.
+		///
+		/// # <weight>
+		/// - `O(1)`
+		/// - 1 event.
+		/// # </weight>
+		#[pallet::weight(T::SystemWeightInfo::remark_with_event(remark.len() as u32))]
+		pub(crate) fn remark_with_event(origin: OriginFor<T>, remark: Vec<u8>) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+			Self::deposit_event(Event::Remarked(remark));
```

Comment:
I still do not see what this ACTUALLY does.

Like events should be for presenting data which may not easily be accessible, for example that an extrinsic completed using fork path A versus fork path B.

In this case, all the information you need is already on chain and accessible as easily as an even it:

* Extrinsic with Remark data is in the block header
* Success event of the remark is in events already

So what am I missing? Is this just lazily trying to use the events API in JavaScript? if so, then we could probably program in the JS API side this functionality.

Reply:
No. Please see my previous comment

> The best practice is always check for specific events, not extrinsic success.

> Otherwise it is very difficult to check if a remark is "executed", think about scheduler, batch, multisig, derive, proxy, delayed proxy and all the combinations and future additions. You simply cannot use extrinsic success to infer if remark is executed without false negative.

> And the main reason I want this is for proof-of-ownership, i.e. for multisgn / proxy account to proof some offchain components someone does control the said account. Otherwise there is no easy way other than transfer some DOT to a random generated account, which is less than ideal.

Reply:
Alice is doing the best practice to secure her account that she have anonymous account + time delayed proxy + revoke proxy + multisig proxy.

Now she want to use a service that require her to provide proof of ownership.

She cannot use her account to sign message as there is no private key for anonymous proxy.

She can use the time delayed proxy to issue a proof as form of system.remark.

Now on the service side, how would you be able to detect if the system.remark is not cancelled?

Reply:
> remark is executed without false negative.

Why do we care about false negative? We should be worried about false positive. In any of the scenarios you describe above, if the full batch/multisig/proxy etc is completed successfully, then the remark is easy enough to verify.

Reply:
Batch will always give extrinsic success. Multisig is requires x success and it is hard to tell which one is the final one. Time delayed proxy execution could fail even if all extrinsic are success. Only check extrinsic success is certainly going to cause some bugs.

And how do I even know if a proxy call contains system.remark? Remember, we can arbitrarily nest calls. batch in proxy in multisig in batch.